text/0000-a11y-routing.md

```diff
@@ -22,66 +22,134 @@ One of the challenges is that screen reader technology is all closed-source exce
+The addon style (popular sr-only technique): 
 
-## How we teach this
+```css
+#ember-a11y-refocus-nav-message {
+  position: absolute;
+  width: 1px;
+  height: 1px;
+  padding: 0;
+  overflow: hidden;
+  clip: rect(0, 0, 0, 0);
+  white-space: nowrap;
+  border: 0;
+}
+```
+
+The addon would attempt to provide a sensible resolution for all involved:
+- the performance gains from `pushState` remain in place
+- users with assistive technology are informed that a page transition has occurred
+- the focus is reset to the message itself, which also resets the focus of the page (as is desired for the screen-reader user)
+- `aria-live` can remain available for things that genuinely need it (a page transition should not need it)
+
+Some [experimentation of this approach can be seen online](https://navigator-message-test-app.netlify.com/). In this example, the message _is_ shown for demonstration purposes, but in the final product, the message would not be shown.  
+
+What do users with screen-readers think? "It works better than I thought it might."
+
+### Solution #2 - Content Focus
+
+What about content-level focus? [Ember-self-focused](https://github.com/linkedin/self-focused/tree/master/packages/ember-self-focused) and [ember-a11y](https://github.com/ember-a11y/ember-a11y) are both examples of community-produced addons that attempt to provide a contextual focus, based on new page content. 
+
+After some research, we discovered that we could provide content-level focus by using (currently private) API to place focus around the content of the application. 
 
-For phase one, the guides should include information about skip links, since the application author will need to add those in themselves (just as they would for a static site). For phase two, we would explain that skip links are no longer needed because we have integrated that functionality in a more machine intelligent way. 
+For the template, we could take one of two approaches: 
 
-### Skip Links
-A skip link is an internal link at the beginning of a hypertext document that permits users to skip navigational material and quickly access the document's main content. Skip links are particularly useful for users who access a document with screen readers and users who rely on keyboards.
+1. wrap the `{{outlet}}` on application.hbs with an element with the tabindex and id attributes, like this:
 
-Directly inside the `body` element, place a skip link like this: 
+```hbs
+<main tabindex="-1" id="ember-primary-application-outlet">
+  {{outlet}}
+</main>
+```
 
-    <a href="#maincontent">Skip to main content</a> 
+2. change the application.hbs outlet so it is different from the usual `{{outlet}}`, and have it already include what we need:
 
-This allows the user with assistive technology to skip directly to the main content and is useful for not needing to repeatedly navigate through a navbar, for example.  
+```hbs
+{{application-outlet}}
```

Comment:
I much prefer having a single `outlet` type. @ef4 and I have created a few over time:
- `liquid-outlet` ... from `liquid-fire`.
- `focusing-outlet` ... from `ember-a11y` which lifts the outlet stuff from `liquid-fire`.
- `animated-outlet` ... in Apple Music which is a few dozen lines of code for hardcoded CSS fade transitions.

The problem with this strategy is that none of those solutions by default play nicely together: they're fighting over the same extension point. To wire them together you have to make one dependent upon another and wrap things. You end up (approximately) with:

```hbs
{{! custom-outlet.hbs }}
{{#animated-outlet}}
  {{#focusing-outlet}}
    {{outlet}}
  {{/focusing-outlet}}
{{/animated-outlet}}
```

I feel like the API for the `outlet` itself should be extensible instead of needing to wrap it. I'm thinking something like element modifiers. It's a little hard to figure out what it should do since and `outlet` doesn't exactly have the same lifecycle hooks that a component has.